run.go

```diff
@@ -0,0 +1,149 @@
+package codacytool
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// RunConfiguration contains the process run configuration
+type RunConfiguration struct {
+	SourceDir            string
+	ToolConfigurationDir string
+	Timeout              time.Duration
+	Debug                bool
+}
+
+const (
+	defaultTimeout              = 15 * time.Minute
+	defaultDebug                = false
+	defaultSourceDir            = "/src"
+	defaultToolConfigurationDir = "/"
+)
+
+// StartTool receives the tool implementation as parameter and runs the tool.
+// Issues found will be printed to the standard output in a JSON format.
+//
+// Return codes are as follows:
+//   - 0 - Tool executed successfully
+//   - 1 - An unknown error occurred while running the tool
+//   - 2 - Execution timeout
```

Comment:
https://github.com/codacy/codacy-example-tool/tree/09538d9fda88e448e58d9599542ac3e028f0075e#behavior-of-the-configuration-file